mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutor.java

```diff
@@ -413,12 +417,14 @@ public void runOneIteration() throws Exception {
                     })
                     .get(heartBeatTimeout.getSize(), heartBeatTimeout.getUnit());
 
-                // the heartbeat was successful, let's reset the counter.
+                // the heartbeat was successful, let's reset the counter and set the registered flag
                 numFailedHeartbeats = 0;
+                registered = true;
             } catch (Exception e) {
                 log.error("Failed to send heartbeat to gateway {}", gateway, e);
-                // increase the number of failed heartbeats by 1
+                // increase the number of failed heartbeats by 1 and clear the registered flag
                 numFailedHeartbeats += 1;
+                registered = false;
```

Comment:
this feels a bit weird: setting this variable to false for every hearthbeat failure, is that what you want?

Reply:
Good point.